File: 238-product-of-array-except-self/product-of-array-except-self.java
class Solution 
{
    public int[] productExceptSelf(int[] nums) 
    {
        int len = nums.length;
        int[] result = new int[len];
        int lpc = 1;

        // Left Product Calculation (LPC)
        result[0] = 1;
        for (int i = 1; i < len; i++) 
        {
            lpc *= nums[i-1];
            result[i] = lpc;
        }
        
        // Right Procduct Calculation (RPC)
        int rpc = 1;
        for(int i = len-1; i>=0; i--)
        {
            result[i] *= rpc;
            rpc *= nums[i];
        }

        return result;
    }
}

File: 238-product-of-array-except-self/product-of-array-except-self.cs
public class Solution 
{
    public int[] ProductExceptSelf(int[] nums) 
    {
        int len = nums.Length;
        int [] answer = new int [len];
        answer[0] = 1;
        
        int lpc = 1;
        for(int i = 1; i<len; i++)
        {
            lpc *= nums[i-1];
            answer[i] = lpc;
        }
        
        int rpc = 1;
        for(int i = len-1; i > 0; i--)
        {
            rpc *= nums[i];
            answer[i-1] *= rpc; 
        }

        return answer;
    }
}